### DISCUSSION TITLE
Implementing a recently published algorithm for diagonalizing Pauli operators

### DISCUSSION THREAD

**Discussion by emm71201** [Qiskit/.github#33] [Ideas]
My colleague, Michael, and I just submitted this paper on the ArXiv. We developed an algorithm to compile a quantum circuit to simultaneously diagonalize commuting Pauli operators (Pauli strings). Remarkably, this algorithm guarantees circuit depth only O(n log n), which is the best scaling as far as we know. We have this algorithm implemented and we want to make it available in Qiskit. Here is the link to the paper: https://arxiv.org/abs/2306.00170.

**Comment by javabster** [comment]
hi @emm71201 thanks for the suggestion! This could be something potentially interesting to include in qiskit at some point in the future, but we would need to see the fully working code with some tests and documentation before deciding if it is an appropriate fit. We recommend for large feature suggestions like this that contributors first open source the work themselves and join the [Qiskit Ecosystem](https://github.com/qiskit-community/ecosystem#ecosystem), and then the feature can be useful on it's own first and lays the groundwork for potential integration into qiskit itself later on.

**Reply by emm71201** [reply]
Hi Abby, thank you for your reply. I will make some changes on the codes before open-sourcing and joining the Qiskit Ecosystem. 
-- Edison
